rcsb/workflow/bcif/task_functions.py

```diff
@@ -0,0 +1,414 @@
+        procs.clear()
+
+    try:
+        for dtemp in dtemps:
+            if os.path.exists(dtemp):
+                shutil.rmtree(dtemp)
+        shutil.rmtree(tempPath)
+    except Exception as e:
+        logger.error(str(e))
+
+    return True
+
+
+def splitList(nfiles: int, subtasks: int, tasklist: List[str]) -> List[List[str]]:
+    step = nfiles // subtasks
+    if step < 1:
+        step = 1
+    steps = nfiles // step
+    logger.info("splitting %d files into %d steps with step %d", nfiles, steps, step)
+    if not isinstance(tasklist[0], str):
+        tasklist = [str(task) for task in tasklist]
+    tasks = [
+        (
+            tasklist[index * step : step + index * step]
+            if index < steps - 1
+            else tasklist[index * step : nfiles]
+        )
+        for index in range(0, steps)
+    ]
+    return tasks
+
+
+def batchTask(
+    tasks,
+    remotePath,
+    updateBase,
+    outfileSuffix,
+    contentType,
+    outputContentType,
+    outputHash,
+    dictionaryApi,
+    tempPath,
+    dtemp,
+    maxTempFiles,
+):
+    logger.info("processing %d tasks", len(tasks))
+    for task in tasks:
+        singleTask(
+            task,
+            remotePath,
+            updateBase,
+            outfileSuffix,
+            contentType,
+            outputContentType,
+            outputHash,
+            dictionaryApi,
+            tempPath,
+            dtemp,
+            maxTempFiles,
+        )
+
+
+def singleTask(
+    pdbId,
+    remotePath,
+    updateBase,
+    outfileSuffix,
+    contentType,
+    outputContentType,
+    outputHash,
+    dictionaryApi,
+    tempPath,
+    dtemp,
+    maxTempFiles,
+    counter=[0],
+):
+    if contentType == "pdb":
+        pdbId = pdbId.lower()
+    elif contentType == "csm":
+        pdbId = pdbId.upper()
+    remoteFileName = "%s%s" % (
+        pdbId,
+        outfileSuffix.replace(".bcif.gz", ".cif.gz").replace(".bcif", ".cif"),
+    )
+    # copy or download to cifFilePath
+    if not remotePath.startswith("http"):
+        # local file
+        localFilePath = os.path.join(remotePath, remoteFileName)
+        if not os.path.exists(localFilePath):
+            logger.error("%s not found", localFilePath)
+            return
+        cifFilePath = os.path.join(tempPath, remoteFileName)
+        shutil.copy(localFilePath, cifFilePath)
+    else:
+        url = os.path.join(remotePath, pdbId[-3:-1], remoteFileName)
+        cifFilePath = os.path.join(tempPath, remoteFileName)
+        if contentType == "csm":
+            url = os.path.join(
+                remotePath, pdbId[0:2], pdbId[-6:-4], pdbId[-4:-2], remoteFileName
+            )
+            cifFilePath = os.path.join(tempPath, remoteFileName)
+        try:
+            r = requests.get(url, timeout=300, stream=True)
+        except requests.exceptions.ConnectTimeout as e:
+            logger.exception(str(e))
+            return
+        try:
+            if r and r.status_code < 400:
```

Comment:
I think here we want to be more strict:
```suggestion
            if r and r.status_code != 200:
```